WordPress/gutenberg
# Ability to rename pages in site editor without title block

<img width="1510" alt="image" src="https://github.com/WordPress/gutenberg/assets/1072756/b6a44e6d-fcf5-4278-af78-e3a92c1cb83a">

## What?
Adds ability to rename pages when in site editor when no post title block exists in template. This PR also adds rename menu item to templates as well to be consistent.

## Why?
https://github.com/WordPress/gutenberg/issues/52763

## How?
Updates existing rename menu item component for templates to be used for any post type.

## Testing Instructions
1. Open a page in the site editor
2. Open page inspector and click more menu -> rename
3. Edit name of page and notice it updates
4. Do the same in the details panel in site editor (left nav)
